homework/index.js

```diff
@@ -27,21 +27,159 @@
         elem.setAttribute(key, value);
       }
     });
+
     return elem;
   }
 
+  function sortList() {
+    let i;
+    let switching;
+    let b;
+    let shouldSwitch;
+    const list = document.getElementById('select');
+    switching = true;
+    while (switching) {
+      switching = false;
+      b = list.getElementsByTagName('option');
+      for (i = 0; i < b.length - 1; i++) {
+        shouldSwitch = false;
+        if (b[i].innerHTML.toLowerCase() > b[i + 1].innerHTML.toLowerCase()) {
+          shouldSwitch = true;
+          break;
+        }
+      }
+      if (shouldSwitch) {
+        b[i].parentNode.insertBefore(b[i + 1], b[i]);
+        switching = true;
+      }
+    }
+  }
+
+  // fetch url contributor
+  function fetchContributorsUrl(contributorsUrl) {
+    fetchJSON(contributorsUrl, (err, contributor) => {
+      if (err) {
+        const root = document.getElementById('root');
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        for (let j = 0; j < contributor.length; j++) {
+          const nameOfContributor = contributor[j].login;
+          const numberOfContributor = contributor[j].contributions;
+          const avatarUrl = contributor[j].avatar_url;
+          const cont = document.getElementById('cont');
+          const avatarContainer1 = createAndAppend('div', cont, {
+            class: 'avatar_div',
+          });
+          const avatarContainer2 = createAndAppend('div', cont, { class: 'avatar_div data' });
+
+          createAndAppend('img', avatarContainer1, { src: avatarUrl });
+          createAndAppend('a', avatarContainer2, {
+            text: nameOfContributor.toUpperCase(),
+            target: '_blank',
+            href: `https://github.com/${nameOfContributor}`,
+          });
+          createAndAppend('br', avatarContainer2, {});
+
+          createAndAppend('span', avatarContainer2, {
+            text: 'Forks: ',
+          });
+          createAndAppend('span', avatarContainer2, {
+            text: numberOfContributor,
+            class: 'result label',
+          });
+          createAndAppend('br', avatarContainer2, {});
+        }
+      }
+    });
+  }
+
+  function reloadSelectedRepository(valueOfSelectedRepository, selectedData) {
```

Comment:
You could pass the data only, instead of passing the array and index.

Reply:
Which data do you mean here? I used data by many functions to have access to scope a another which means don't need to use closure functions. Does that make sense?
1- scoops 
2-callbacks: call function inside another instead of using closure functions.
I am actually not sure about those cases but i am trying with my code to get the whole image. If this explanation not the answer to you question, I hope to hear from you about that.

Thanks

Reply:
What I meant by data is the repo JSON data, where you change the function signature to 

    function reloadSelectedRepository(repo)

So `selectedData[valueOfSelectedRepository]` will be replaced with `repo`

And the call in https://github.com/Muhannad1989/JavaScript3/blob/d9aba0940f5dd5bdddc82c3e2e5fb4b3a167df0d/homework/index.js#L159 will be changed to `reloadSelectedRepository(data[0]);` and the same for https://github.com/Muhannad1989/JavaScript3/blob/d9aba0940f5dd5bdddc82c3e2e5fb4b3a167df0d/homework/index.js#L164 the function call will be changed to `reloadSelectedRepository(data[indexOfSelectedRepo]);`

Reply:
I did it , it has now only one parameter. Thanks